sirepo/template/code_variable.py

```diff
@@ -63,6 +66,29 @@ def eval_var_with_assert(self, expr):
         assert not err, err
         return float(v)
 
+    def get_application_data(self, args, schema, ignore_array_values=False):
+        from sirepo import simulation_db
+        if args.method == 'rpn_value':
+            if ignore_array_values and re.search(r'^\{.*\}$', args.value):
+                # accept array of values enclosed in curly braces
+                args.result = ''
+                return True
+            v, err = self.eval_var(args.value)
+            if err:
+                args.error = err
+            else:
+                args.result = v
+            return True
+        if args.method == 'recompute_rpn_cache_values':
+            self.recompute_cache(args.cache)
+            return True
+        if args.method == 'validate_rpn_delete':
+            model_data = simulation_db.read_json(
```

Comment:
This is problematic. It bypasses the concept of validating the simulationId and simulationType. It also decouples the values that are in the GUI at the time get_application_data is called from the values on disk on the server, which could have been saved after those values were first retrieved. This is a known problem elsewhere, but it seems like we are continually avoiding the problem that what the GUI things of the model is different from what is in the db. If we passed the serial, it would guarantee that the two are in agreement.